### ISSUE TITLE
Documentation for push server is not a trusted proxy by Nextcloud

### ISSUE THREAD

**Issue by wmutschl** [linuxserver/docker-mods#903]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Name of mod

nextcloud-notify-push

### Name of base container

nextcloud

### Current Behavior

I get the following in my logs and do not know what to do:

```
nextcloud  | ───────────────────────────────────────
nextcloud  | 
nextcloud  |       ██╗     ███████╗██╗ ██████╗
nextcloud  |       ██║     ██╔════╝██║██╔═══██╗
nextcloud  |       ██║     ███████╗██║██║   ██║
nextcloud  |       ██║     ╚════██║██║██║   ██║
nextcloud  |       ███████╗███████║██║╚██████╔╝
nextcloud  |       ╚══════╝╚══════╝╚═╝ ╚═════╝
nextcloud  | 
nextcloud  |    Brought to you by linuxserver.io
nextcloud  | ───────────────────────────────────────
nextcloud  | 
nextcloud  | To support LSIO projects visit:
nextcloud  | https://www.linuxserver.io/donate/
nextcloud  | 
nextcloud  | ───────────────────────────────────────
nextcloud  | GID/UID
nextcloud  | ───────────────────────────────────────
nextcloud  | 
nextcloud  | User UID:    1000
nextcloud  | User GID:    911
nextcloud  | ───────────────────────────────────────
nextcloud  | 
nextcloud  | using keys found in /config/keys
nextcloud  | Initializing nextcloud 29.0.0.19 (this can take a while) ...
nextcloud  | Setting permissions
nextcloud  | Initializing finished
nextcloud  | [custom-init] No custom files found, skipping...
nextcloud  | **** Making sure the Nextcloud Client Push plugin is installed and enabled ****
nextcloud  | notify_push already installed
nextcloud  | notify_push already enabled
nextcloud  | **** Starting notify-push ****
nextcloud  | Connection to localhost (127.0.0.1) 7867 port [tcp/*] succeeded!
nextcloud  | **** Setting notify_push server URL to https://nextcloud.conraths.net//push ****
nextcloud  | ✓ redis is configured
nextcloud  | ✓ push server is receiving redis messages
nextcloud  | ✓ push server can load mount info from database
nextcloud  | ✓ push server can connect to the Nextcloud server
nextcloud  | 🗴 push server is not a trusted proxy by Nextcloud or another proxy in the chain.
nextcloud  |   Nextcloud resolved the following client address for the test request: "::1" instead of the expected "1.2.3.4"
nextcloud  |   The following trusted proxies are currently configured: "172.30.0.3", "127.0.0.1"
nextcloud  |   The following x-forwarded-for header was received by Nextcloud: "1.2.3.4"
nextcloud  |     from the following remote: ::1
nextcloud  | 
nextcloud  |   ::1 is not a trusted as a reverse proxy by Nextcloud
nextcloud  |   See https://docs.nextcloud.com/server/latest/admin_manual/configuration_server/reverse_proxy_configuration.html#defining-trusted-proxies for how to add trusted proxies.
nextcloud  | 
nextcloud  |   If you're having issues getting the trusted proxy setup working, you can try bypassing any existing reverse proxy
nextcloud  |   in your setup by setting the `NEXTCLOUD_URL` environment variable to point directly to the internal Nextcloud webserver url
nextcloud  |   (You will still need the ip address of the push server added as trusted proxy)
nextcloud  | **** There was an error setting the notify_push server URL. Please double check your reverse proxy settings as well as the overwrite.cli.url entry in Nextcloud's config.php ****
nextcloud  | [ls.io-init] done.
```

### Expected Behavior

What exactly should I add to my config file under trusted_proxy as I am running this with the swag container?

### Steps To Reproduce

My docker-compose file:

```
version: "3"

networks:
  nextcloud_network:
    name: nextcloud_network

services:
  swag:
    image: lscr.io/linuxserver/swag:2.10.0-ls302
    container_name: swag
    cap_add:
      - NET_ADMIN
    environment:
      - PUID=1000
      - PGID=1000
      - TZ=${TIMEZONE}
      - URL=${SWAG_URL}
      - SUBDOMAINS=${SWAG_SUBDOMAINS}
      - EXTRA_DOMAINS=${SWAG_EXTRA_DOMAINS}
      - VALIDATION=${SWAG_VALIDATION}
      - DNSPLUGIN=${SWAG_DNSPLUGIN}
      - EMAIL=${EMAIL}
      - CERTPROVIDER=${SWAG_CERTPROVIDER}
    volumes:
      - ${SWAG_ROOT}:/config
    ports:
      - 443:443
      - 80:80
    restart: unless-stopped
    networks:
      - nextcloud_network

  nextcloud:
    image: lscr.io/linuxserver/nextcloud:29.0.0-ls321
    container_name: nextcloud
    environment:
      - PUID=1000
      - GUID=1000
      - TZ=${TIMEZONE}
      - DOCKER_MODS=linuxserver/mods:nextcloud-notify-push
    volumes:
      - ${NEXTCLOUD_ROOT}/config:/config
      - ${NEXTCLOUD_ROOT}/data:/data
    depends_on:
      - nextcloud-mariadb
      - nextcloud-redis
    restart: unless-stopped
    networks:
      - nextcloud_network

  nextcloud-mariadb:
    image: lscr.io/linuxserver/mariadb:10.11.6-r0-ls141
    container_name: nextcloud-mariadb
    environment:
      - PUID=1000
      - GUID=1000
      - TZ=${TIMEZONE}
      - MYSQL_DATABASE=${MYSQL_DATABASE}
      - MYSQL_USER=${MYSQL_USER}
      - MYSQL_PASSWORD=${MYSQL_PASSWORD}
      - MYSQL_ROOT_PASSWORD=${MYSQL_ROOT_PASSWORD}
    volumes:
      - ${NEXTCLOUD_ROOT}/mariadb:/config
    restart: unless-stopped
    networks:
      - nextcloud_network

  nextcloud-redis:
    image: redis:7.2.5-bookworm
    container_name: nextcloud-redis
    command: redis-server --requirepass ${REDIS_PASSWORD}
    restart: unless-stopped
    networks:
      - nextcloud_network



### Environment

```markdown
- OS: Ubuntu 20.04
- How docker service was installed: following the official guide by Docker
```

### CPU architecture

x86-64

### Docker creation

```bash
docker compose up -d
```

### Container logs

```bash
nextcloud  | ───────────────────────────────────────
nextcloud  | 
nextcloud  |       ██╗     ███████╗██╗ ██████╗
nextcloud  |       ██║     ██╔════╝██║██╔═══██╗
nextcloud  |       ██║     ███████╗██║██║   ██║
nextcloud  |       ██║     ╚════██║██║██║   ██║
nextcloud  |       ███████╗███████║██║╚██████╔╝
nextcloud  |       ╚══════╝╚══════╝╚═╝ ╚═════╝
nextcloud  | 
nextcloud  |    Brought to you by linuxserver.io
nextcloud  | ───────────────────────────────────────
nextcloud  | 
nextcloud  | To support LSIO projects visit:
nextcloud  | https://www.linuxserver.io/donate/
nextcloud  | 
nextcloud  | ───────────────────────────────────────
nextcloud  | GID/UID
nextcloud  | ───────────────────────────────────────
nextcloud  | 
nextcloud  | User UID:    1000
nextcloud  | User GID:    911
nextcloud  | ───────────────────────────────────────
nextcloud  | 
nextcloud  | using keys found in /config/keys
nextcloud  | Initializing nextcloud 29.0.0.19 (this can take a while) ...
nextcloud  | Setting permissions
nextcloud  | Initializing finished
nextcloud  | [custom-init] No custom files found, skipping...
nextcloud  | **** Making sure the Nextcloud Client Push plugin is installed and enabled ****
nextcloud  | notify_push already installed
nextcloud  | notify_push already enabled
nextcloud  | **** Starting notify-push ****
nextcloud  | Connection to localhost (127.0.0.1) 7867 port [tcp/*] succeeded!
nextcloud  | **** Setting notify_push server URL to https://nextcloud.conraths.net//push ****
nextcloud  | ✓ redis is configured
nextcloud  | ✓ push server is receiving redis messages
nextcloud  | ✓ push server can load mount info from database
nextcloud  | ✓ push server can connect to the Nextcloud server
nextcloud  | 🗴 push server is not a trusted proxy by Nextcloud or another proxy in the chain.
nextcloud  |   Nextcloud resolved the following client address for the test request: "::1" instead of the expected "1.2.3.4"
nextcloud  |   The following trusted proxies are currently configured: "172.30.0.3", "127.0.0.1"
nextcloud  |   The following x-forwarded-for header was received by Nextcloud: "1.2.3.4"
nextcloud  |     from the following remote: ::1
nextcloud  | 
nextcloud  |   ::1 is not a trusted as a reverse proxy by Nextcloud
nextcloud  |   See https://docs.nextcloud.com/server/latest/admin_manual/configuration_server/reverse_proxy_configuration.html#defining-trusted-proxies for how to add trusted proxies.
nextcloud  | 
nextcloud  |   If you're having issues getting the trusted proxy setup working, you can try bypassing any existing reverse proxy
nextcloud  |   in your setup by setting the `NEXTCLOUD_URL` environment variable to point directly to the internal Nextcloud webserver url
nextcloud  |   (You will still need the ip address of the push server added as trusted proxy)
nextcloud  | **** There was an error setting the notify_push server URL. Please double check your reverse proxy settings as well as the overwrite.cli.url entry in Nextcloud's config.php ****
nextcloud  | [ls.io-init] done.
```

**Comment by j0nnymoe** [comment]
I think that error is saying your trusted proxies in your config.php isn't set correctly. Also side note your PGID env for nextcloud is incorrect.

**Comment by wmutschl** [comment]
Now, should I add "::1" to my trusted proxies?

**Comment by wmutschl** [comment]
My 'trusted_proxies' settings is:

```
'trusted_proxies' => [gethostbyname('swag')],
```
All is good.

If I enable notify_push, then this gets changed to:
```
'trusted_proxies' => 
  array (
    0 => '172.28.0.4',
    1 => '127.0.0.1',
  ),
  ```
  Now if I add `::1` as the 2 entry, then it works, but where is this documented?

**Comment by wmutschl** [comment]
Regarding "Also side note your PGID env for nextcloud is incorrect": How can I fix this?

**Comment by j0nnymoe** [comment]
> My 'trusted_proxies' settings is:
> 
> ```
> 'trusted_proxies' => [gethostbyname('swag')],
> ```
> 
> All is good.
> 
> If I enable notify_push, then this gets changed to:
> 
> ```
> 'trusted_proxies' => 
>   array (
>     0 => '172.28.0.4',
>     1 => '127.0.0.1',
>   ),
> ```
> 
> Now if I add `::1` as the 2 entry, then it works, but where is this documented?

Try
`  array (
    0 => '172.28.0.0/12',
    1 => '127.0.0.1',
  ),`

**Comment by j0nnymoe** [comment]
> Regarding "Also side note your PGID env for nextcloud is incorrect": How can I fix this?

By fixing the typo ;)

**Comment by aptalca** [comment]
Did you enable ipv6 in that docker network?

**Comment by wmutschl** [comment]
> Try ` array ( 0 => '172.28.0.0/12', 1 => '127.0.0.1', ),`
did not work. However, this works for me:

```sh
'trusted_proxies' => 
  array (
    0 => '172.28.0.4',
    1 => '127.0.0.1',
    2 => '::1',
  ),
  ```
  So I guess it has to do with ipv6, as `::1` is the IPv6 loopback address which refers to the local machine but in the IPv6 format.

**Comment by wmutschl** [comment]
> Did you enable ipv6 in that docker network?

I don't think I did. My docker-compose file is the following:
```
version: "3"

networks:
  nextcloud_network:
    name: nextcloud_network

services:
  swag:
    image: lscr.io/linuxserver/swag:2.10.0-ls302
    container_name: swag
    cap_add:
      - NET_ADMIN
    environment:
      - PUID=1000
      - PGID=1000
      - TZ=${TIMEZONE}
      - URL=${SWAG_URL}
      - SUBDOMAINS=${SWAG_SUBDOMAINS}
      - EXTRA_DOMAINS=${SWAG_EXTRA_DOMAINS}
      - VALIDATION=${SWAG_VALIDATION}
      - DNSPLUGIN=${SWAG_DNSPLUGIN}
      - EMAIL=${EMAIL}
      - CERTPROVIDER=${SWAG_CERTPROVIDER}
    volumes:
      - ${SWAG_ROOT}:/config
    ports:
      - 443:443
      - 80:80
    restart: unless-stopped
    networks:
      - nextcloud_network
      - gitea_network

  nextcloud:
    image: lscr.io/linuxserver/nextcloud:29.0.0-ls321
    container_name: nextcloud
    environment:
      - PUID=1000
      - PGID=1000
      - TZ=${TIMEZONE}
      - DOCKER_MODS=linuxserver/mods:nextcloud-notify-push
    volumes:
      - ${NEXTCLOUD_ROOT}/config:/config
      - ${NEXTCLOUD_ROOT}/data:/data
    depends_on:
      - nextcloud-mariadb
      - nextcloud-redis
    restart: unless-stopped
    networks:
      - nextcloud_network

  nextcloud-mariadb:
    image: lscr.io/linuxserver/mariadb:10.11.6-r0-ls141
    container_name: nextcloud-mariadb
    environment:
      - PUID=1000
      - PGID=1000
      - TZ=${TIMEZONE}
      - MYSQL_DATABASE=${MYSQL_DATABASE}
      - MYSQL_USER=${MYSQL_USER}
      - MYSQL_PASSWORD=${MYSQL_PASSWORD}
      - MYSQL_ROOT_PASSWORD=${MYSQL_ROOT_PASSWORD}
    volumes:
      - ${NEXTCLOUD_ROOT}/mariadb:/config
    restart: unless-stopped
    networks:
      - nextcloud_network

  nextcloud-redis:
    image: redis:7.2.5-bookworm
    container_name: nextcloud-redis
    command: redis-server --requirepass ${REDIS_PASSWORD}
    restart: unless-stopped
    networks:
      - nextcloud_network

  gitea:
    image: gitea/gitea:1.21.11
    container_name: gitea
    environment:
      - USER_UID=1000
      - USER_GID=1000
      - DOMAIN=${GITEA_DOMAIN}
      - RUN_MODE=prod
      - LFS_START_SERVER=true
      - GITEA__repository__DEFAULT_BRANCH=main
      - GITEA__oauth2__JWT_SECRET=${GITEA_JWT_SECRET}
      - GITEA__security__SECRET_KEY=${GITEA_SECRET_KEY}
      - GITEA__security__INTERNAL_TOKEN=${GITEA_INTERNAL_TOKEN}
      - GITEA__security__INSTALL_LOCK=true
      - GITEA__server__DISABLE_SSH=true
      - GITEA__server__LFS_JWT_SECRET=${GITEA_LFS_JWT_SECRET}
      - GITEA__server__ROOT_URL=https://${GITEA_DOMAIN}
      - GITEA__service__DISABLE_REGISTRATION=true
    restart: unless-stopped
    volumes:
      - ${GITEA_ROOT}:/data
      - /etc/timezone:/etc/timezone:ro
      - /etc/localtime:/etc/localtime:ro
    networks:
      - gitea_network
```

**Comment by wmutschl** [comment]
Also a `ping6` to my nextcloud yields:
```sh
wmutschl@macbook-pro-lan ~> ping6 nextcloud.mutschler.eu -n -c 4
ping6: getaddrinfo -- nodename nor servname provided, or not known

wmutschl@macbook-pro-lan ~ [1]> ping nextcloud.mutschler.eu -n -c 4
PING nextcloud.mutschler.eu (134.2.37.233): 56 data bytes
64 bytes from 134.2.37.233: icmp_seq=0 ttl=55 time=27.375 ms
64 bytes from 134.2.37.233: icmp_seq=1 ttl=55 time=24.514 ms
64 bytes from 134.2.37.233: icmp_seq=2 ttl=55 time=24.482 ms
64 bytes from 134.2.37.233: icmp_seq=3 ttl=55 time=24.484 ms
```

**Comment by ntompson** [comment]
Adding to this - I've also just activated the push server, per current documentation. Unless I add "::1" as noted above, push doesn't work at all. The logic is a bit lost on me as to why this should be necessary. 

> > Try ` array ( 0 => '172.28.0.0/12', 1 => '127.0.0.1', ),`
> 
> did not work. However, this works for me:
> 
> ```shell
> 'trusted_proxies' => 
>   array (
>     0 => '172.28.0.4',
>     1 => '127.0.0.1',
>     2 => '::1',
>   ),
> ```
> 
> So I guess it has to do with ipv6, as `::1` is the IPv6 loopback address which refers to the local machine but in the IPv6 format.

**Comment by aptalca** [comment]
Somehow you guys must have enabled (or docker did) ipv6 in your user defined bridge network and when notify-push tries to connect Nextcloud via `localhost`, Nextcloud sees the request coming from `::1` instead of `127.0.0.1`.

I opened a PR to add both localhost addresses to trusted proxies. Please give it a try and let us know.

Just replace `linuxserver/mods:nextcloud-notify-push` with `ghcr.io/linuxserver/mods:pull_request_905`

Here's the PR: https://github.com/linuxserver/docker-mods/pull/905

**Comment by ntompson** [comment]
Interesting. I haven't done anything specific to *enable* IPv6, but I presume that docker supports IPv6 out of the box. I tried adding `enable_ipv6: false` to the various docker networks involved, but that didn't seem to do anything. Adding `::1` to `trusted_proxies` certainly works, but I am curious to understand why IPv6 has been brought into the mix in the first place.

**Comment by aptalca** [comment]
Fixed in #905
